src/public/app/core/types/ScPlayer.ts

```diff
@@ -9,20 +9,20 @@ export interface ScPlayer {
 }
 
 export enum ScPlayerEvent {
-    StateChange = 'state-change',
-    Play = 'play',
-    PlayStart = 'play-start',
-    PlayResume = 'play-resume',
-    Pause = 'pause',
-    Finished = 'finish',
-    Seek = 'seek',
-    Seeked = 'seeked',
-    GeoBlocked = 'geo_blocked',
-    BufferingStart = 'buffering_start',
-    BufferingEnd = 'buffering_end',
-    AudioError = 'audio_error',
-    Time = 'time',
-    NoStreams = 'no_streams',
-    NoProtocol = 'no_protocol',
-    NoConnection = 'no_connection'
+    StateChange = <any>'state-change',
```

Comment:
the strings are whats passed to the sc object.
https://developers.soundcloud.com/docs/api/sdks#stream I wanted a way for
typescript to tell me all of the options

On Tue, Apr 19, 2016 at 6:04 PM, dswen <notifications@github.com> wrote:

> In src/public/app/core/types/ScPlayer.ts
> <https://github.com/ryanquinn3/beetz/pull/8#discussion_r60337502>:
>
> > -    Play = 'play',
> > -    PlayStart = 'play-start',
> > -    PlayResume = 'play-resume',
> > -    Pause = 'pause',
> > -    Finished = 'finish',
> > -    Seek = 'seek',
> > -    Seeked = 'seeked',
> > -    GeoBlocked = 'geo_blocked',
> > -    BufferingStart = 'buffering_start',
> > -    BufferingEnd = 'buffering_end',
> > -    AudioError = 'audio_error',
> > -    Time = 'time',
> > -    NoStreams = 'no_streams',
> > -    NoProtocol = 'no_protocol',
> > -    NoConnection = 'no_connection'
> > +    StateChange = <any>'state-change',
>
> Or just scrap the strings altogether. Is there a reason you added strings
> to go along? Why not just
> enum { StateChange, Play, PlayStart, ... }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/ryanquinn3/beetz/pull/8/files/2aabd0b9fad287cfc337421e70eb4c4f70cfdaed#r60337502>
>